include/tracing/tracing.h

```diff
@@ -132,6 +132,41 @@
  */
 #define sys_trace_idle()
 
+/**
+ * @brief Trace initialisation of a Semaphore
+ * @param sem Semaphore object
+ */
+#define sys_trace_semaphore_init(sem)
```

Comment:
what do these empty definitions do? is this for when tracing is disabled? I don't think we would need doxygen here.

Reply:
this for when tracing is disabled, doxygen actually belongs here and not in the implementation.